lib/Analysis/Allocation.cpp

```diff
@@ -185,15 +210,56 @@ class AllocationAnalysis {
   using BufferT = Allocation::BufferT;
 
   /// Value -> Liveness Range
+  using IntervalT = Interval<size_t>;
   /// Use MapVector to ensure determinism.
-  using BufferRangeMapT = llvm::MapVector<BufferT *, Interval<size_t>>;
+  using BufferRangeMapT = llvm::MapVector<BufferT *, IntervalT>;
   /// Nodes -> Nodes
   using GraphT = DenseMap<BufferT *, DenseSet<BufferT *>>;
 
+  /// Set of Liveness Intervals
+  class LiveIntervals : public SmallVector<IntervalT, 4> {
```

Comment:
I agree, this is confusing

Reply:
This is following a IS-A relationship (vs HAS-A) since it represents a set of Intervals, with a few couple methods to simplify. 
I can make it a HAS-A, but will have to override iterator methods (begin, end), not a big deal.

Reply:
yes this would be much easier to read and avoid inheriting all the SmallVector methods which exposes much more than you want to.